First draft of process documentation

joeyparrish opened the pull request.

joeyparrish committed:
First draft of process documentation

joeyparrish commented:
You can see a preview of this here: https://joeyparrish.github.io/shaka-project.github.io/

theodab commented on maintenance/README.md:
```diff
@@ -0,0 +1,164 @@
+longer weekly sessions) to keep response times low.
+
+We use an open-source tool called
+[Triage Party](https://triage-party.shakalab.rocks/) to triage GitHub issues
```
I feel like this link is slightly misleading? I would expect this to link to, like, the GitHub page for Triage Party, since this sentence is talking about the tool, and for the link to OUR Triage Party to be in a follow-up sentence. Especially since in the other sections, we link to the GitHub page when first discussing the tool.

joeyparrish replied:
Good point.  We link to the specific views right after this already.  Fixed.

theodab commented on maintenance/README.md:
```diff
@@ -0,0 +1,164 @@
+The issues in the triage collections should be investigated in order.  For
+example, "untyped issues" should have their type set (`type: bug`,
+`type: enhancement`, `type: question`, etc.), then "unprioritized issues"
+should have a priority label set (`priority: P1`, `priority: P2`, etc.)
```
Not related to this PR in particular, but I double-checked the tips on Triage Party, and I feel like "If it's an enhancement, it is almost always a P2 or lower" is confusing. Since "or lower" is meant to mean "lower priority", but it could also mean "lower number" (e.g. P2 or P1 or P0).

joeyparrish replied:
I see what you mean.  If you want to make changes to that, see https://github.com/shaka-project/triage-party-config/blob/8e9ab874108b93daa8855fd7039a6c8052718e40/shaka-triage-party-config.yaml#L328

theodab commented on maintenance/README.md:
```diff
@@ -0,0 +1,164 @@
+and the check will be run again.
+
+Maintainers should install the [Squashed Merge Message browser extension](https://github.com/zachwhaley/squashed-merge-message#install)
+to make sure your PR commits match the PR title and description.  The extension
```
You switched from third-person ("maintainers") to second-person ("you") mid-paragraph.

joeyparrish replied:
Fixed

theodab commented on maintenance/shaka-player.md:
```diff
@@ -0,0 +1,180 @@
+For example, in v4.0, we were forced to make a change that disabled HLS support
+for older smart TVs.  In that same release, we removed deprecated methods,
+configurations that were added to maintain compatibility with older releases,
+etc.  Many of them had been marked deprecated for 6-12 months already.
```
I feel like this doc should mention the need to update the upgrade guide docs, for breaking changes.

joeyparrish replied:
Done

joeyparrish committed:
First round feedback

 - fix triage party link
 - fix third-person/second-person switch mid-sentence
 - add notes on upgrade guide

avelad commented on maintenance/shaka-player.md:
```diff
@@ -0,0 +1,185 @@
+
+ 1. Announce the release via email
+
+    > :pencil: **TODO**: Automate this or deprecate the practice!
```
I am in favor of eliminating this practice, people can already subscribe to the GitHub to obtain updates

joeyparrish replied:
Ack.  I already failed to email on the last couple of releases.  I think what I need to do now is email the list and announce the end of email announcements.  :-)  I'll send out instructions for subscribing via GitHub.

joeyparrish replied:
Removed.  I've put up a docs PR for Shaka Player to change the way we describe the mailing list, and to link to instructions for subscribing to releases via GitHub.  When that's approved, I'll announce to the list that we're ending announcements.

avelad replied:
Thanks!

joeyparrish replied:
Merged in player, and packager, and announced via email.

avelad commented on maintenance/shaka-player.md:
```diff
@@ -0,0 +1,185 @@
+
+ 1. Update the Chromecast receiver app URLs for the Shaka Player Demo
+
+    > :pencil: **TODO**: Automate this!
```
I just saw that this is very important because when I released v4.1 I got this error without realizing it :(

joeyparrish replied:
Now that I'm looking at this again, I think there are two parts I need to mention.

One is that we need to generate a new app ID for a new feature release for compatibility.  I don't think there's a good way to automate that at the moment.  This will have to be Google's responsibility.

The other is that we need to update the registered URLs for bugfix releases for existing cast app IDs.  This is the part we could potentially "automate" through appspot, by having a URL like "v3-2-x-dot-shaka-player-demo.appspot.com" that would always point to the most recent release for the given branch.  Then that can be the registered URL for the receiver app, and it won't need to be updated in the Cast developer console on bugfix releases.

joeyparrish replied:
Done.

avelad commented on maintenance/shaka-player.md:
```diff
@@ -0,0 +1,185 @@
+   (major release number) will receive fixes for 1 year
+   - _For example, v3.3 (right before v4.0) will be fixed for 1 year past the
+     release of v4.0.0 (until April 30, 2023)._
+ - The release currently used by the Cast Application Framework will continue
```
Here it would be good to define whether the maintainers do the cherry-pick or the Google team does it.

joeyparrish replied:
Yes, definitely.  I think it could reasonably be the responsibility of Googlers to maintain the branch for CAF, since it's not obvious what branch we're on in that closed-source project.

joeyparrish replied:
Done.

avelad commented on maintenance/shaka-player.md:
```diff
@@ -0,0 +1,185 @@
+# Inspect all commits since a particular main-branch tag.
+# If the most recent v3.3.x release was v3.3.1, the tag
+# v3.3.1-main is the same point in history for the main branch.
+interactive-cherry-pick v3.3.1-main upstream main
```
Here I have discovered that only the tag is mandatory, the rest of the parameters have a default value, should it be mentioned here?

joeyparrish replied:
That's true if you set up your local clone the way we recommend.  I wanted to mention the parameters, though, so it's clear that they exist and that you might need to set them.

I'll see if I can find a concise way to explain all that in the doc.

joeyparrish replied:
Done

avelad reviewed (commented):
In general, I see it well, but there is a synchronization thing that I do not see resolved here: How do we know what each one is working on? Perhaps assigning the issues is the best way, or maybe in the Chat mention it. On the other hand, a very useful functionality for users was to know the roadmap, should we keep this?

joeyparrish commented:
> How do we know what each one is working on? Perhaps assigning the issues is the best way, or maybe in the Chat mention it.

I think issue assignment is the best way.  In addition to being built into GitHub and visible to everyone, it will feed into triage-party, which has a list of assigned issues.  Whoever is doing triage can look quickly and see if there is someone who is assigned a bunch of issues and not making progress (_it's me_), and can start a conversation about rebalancing the load.

> On the other hand, a very useful functionality for users was to know the roadmap, should we keep this?

Yes, I think we should keep the roadmap.md file up-to-date if we can.

joeyparrish committed:
second round feedback from Alvaro

joeyparrish commented:
The rendered preview on my fork has been updated: https://joeyparrish.github.io/shaka-project.github.io/maintenance/shaka-player.html

theodab commented on maintenance/shaka-player.md:
```diff
@@ -0,0 +1,205 @@
+   different, in which case [a decision must be made](https://quotecatalog.com/quote/mitch-hedberg-i-write-jokes-f-jpXNBB1)
+   about whether porting the fix to the branch is worth the effort.  The
+   maintainer will have to consider both the severity of the issue and the
+   difficulty of fixing conflicts or reimplementation.
```
In the case of fixing conflicts or reimplementation, does the new version of the change need to be submitted as a pull request?

joeyparrish replied:
If fixing conflicts is trivial, they don't need to be re-reviewed.  When I *think* a conflict is trivial, and I was wrong, the tests usually fail and I found out that way before pushing.

Reimplementation is another story.  I'll try to clarify that here.

joeyparrish replied:
Done.  PTAL!

joeyparrish committed:
Add notes on conflicts and reimplementation during cherry-picking
